src/domains/TypesDomain.js

```diff
@@ -105,8 +112,9 @@ export default class TypesDomain {
   }
 
   joinWith(t: typeof Value): TypesDomain {
+    if (this.isBottom()) return t === EmptyValue ? TypesDomain.bottomVal : new TypesDomain(t);
     let type = this.getType();
-    if (type === t) return this;
+    if (type === t || t instanceof EmptyValue) return this;
```

Comment:
The `|| t instanceof EmptyValue` should be infeasible given the above `isBottom` check.

Reply:
No, if this is not bottom and t is EmptyValue then we are joining not bottom with bottom, which is not bottom.